src/vs/workbench/services/languageDetection/browser/languageDetectionWorkerServiceImpl.ts

```diff
@@ -0,0 +1,254 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, DisposableStore, dispose, IDisposable, toDisposable } from 'vs/base/common/lifecycle';
+import { ILanguageDetectionService } from 'vs/workbench/services/languageDetection/common/languageDetectionWorkerService';
+import { FileAccess } from 'vs/base/common/network';
+import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IModeService } from 'vs/editor/common/services/modeService';
+import { URI } from 'vs/base/common/uri';
+import { isWeb } from 'vs/base/common/platform';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+import { LanguageDetectionSimpleWorker } from 'vs/workbench/services/languageDetection/browser/languageDetectionSimpleWorker';
+import { DefaultWorkerFactory } from 'vs/base/worker/defaultWorkerFactory';
+import { IModelService } from 'vs/editor/common/services/modelService';
+import { IntervalTimer } from 'vs/base/common/async';
+import { SimpleWorkerClient } from 'vs/base/common/worker/simpleWorker';
+
+export class LanguageDetectionService extends Disposable implements ILanguageDetectionService {
+	static readonly enablementSettingKey = 'workbench.editor.untitled.experimentalLanguageDetection';
+
+	_serviceBrand: undefined;
+
+	private _languageDetectionWorkerClient: LanguageDetectionWorkerClient;
+
+	constructor(
+		@IWorkbenchEnvironmentService private readonly _environmentService: IWorkbenchEnvironmentService,
+		@IModeService private readonly _modeService: IModeService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IModelService modelService: IModelService,
+	) {
+		super();
+
+		this._languageDetectionWorkerClient = new LanguageDetectionWorkerClient(
+			modelService,
+			'languageDetectionWorkerService',
+			// TODO: See if it's possible to bundle vscode-languagedetection
+			FileAccess.asBrowserUri('../../../../../../node_modules/@vscode/vscode-languagedetection/dist/lib/index.js', require).toString(true),
```

Comment:
The worker had trouble importing this module so I had to pass a Uri to it. There's gotta be a better way....